server/plugins/com/openrsc/server/plugins/quests/members/touristtrap/TouristTrap.java

```diff
@@ -2785,4 +2795,21 @@ private boolean succeedRate(Player p) {
 			return true;
 		}
 	}
-}
+	private boolean playerArmed(Player p) {
+        for (Item item : p.getInventory().getItems()) {
+        for (int robes : allow) {
+            allowed.add(robes);
```

Comment:
This will add each robe into the array for each item, causing the object to have the robes multiple times.